### DISCUSSION TITLE
Are there plans for running on Raspberry Pi?

### DISCUSSION THREAD

**Discussion by EwoutdBoer** [dotnet/maui#215] [General] [answered]
Are there plans to target Raspberry Pi on the native linux based Pi image?

**Comment by michaelonz** [comment]
Im interested in this also

**Comment by trampster** [comment, marked as answer]
The problem here is that Raspberry Pi runs linux, and .net Maui officially doesn't support linux, It's listed as community supported, but there is no community working on it, and anyone that has tried has either been blocked (told not to work on it until after it's working on windows) or ignored (see pull request #239)

However I to would love to see .net Maui running on the Raspberry Pi

**Reply by EwoutdBoer** [reply]
Thanks for your reply and links. It would be nice indeed to create user interfaces for linux as well. I tried the Uno platform to run on raspberry pi. That is working ok, but while the Ui is rendered using Skia sharp, it has lots of limitations.
I will have a look at the links. I would like to help out with an initial implementation / pilot / tryout.

**Reply by AmSmart** [reply]
I'm interested in the limitations you ran into. I'm trying to decide a framework for a Raspberry Pi project. Your response might be what I need. Thanks.

**Reply by francoistanguay** [reply]
Just also bear in mind Linux has been on rapid development cycles. So experience from 1 month ago might be outdated and many issues might have been fixed. https://platform.uno/uno-platform-for-linux/

**Reply by henzard** [reply]
Any more on this ?

**Comment by sjordanGSS** [comment]
I've heard of people getting Android to run on Raspberry Pis, this would give you a relatively easy access route for Maui as Android has first-class support. I don't know how good Maui support would be for e.g. Raspberry Pi OS, which as others have pointed out is a flavour of Debian Linux.

**Comment by Hrishikesh46779** [comment]
Any update for this now?
I want to run MAUI app on raspberry pi 4 and access all gpio pins on board.

**Comment by sasakrsmanovic** [comment]
Hi @Hrishikesh46779

You can still use .net to create cross platform applications which also run on Linux / Raspberry Pi by using [Uno Platform](https://github.com/unoplatform/uno).

You can see the getting started with Uno and Pi tutorial at [https://platform.uno/docs/articles/guides/raspberry-pi/raspberry-pi-intro.html](https://platform.uno/docs/articles/guides/raspberry-pi/raspberry-pi-intro.html). Also Pete has shown how to get started on YouTube https://www.youtube.com/watch?v=al1lfBkyx6M

Some of our community have been playing with it for a home surveillance project - [https://twitter.com/dotMorten/status/1554314646915407872](https://twitter.com/dotMorten/status/1554314646915407872) , or also his remote Spotify controller player with Pi which even official Raspberry Pi twitter account retweeted. [https://twitter.com/Raspberry_Pi/status/1658850115858079751](https://twitter.com/Raspberry_Pi/status/1658850115858079751)

**Comment by WanftMoon** [comment]
there is also avalonia if you want the xaml: https://docs.avaloniaui.net/docs/next/guides/platforms/rpi/running-on-raspbian-lite-via-drm

**Comment by tscholze** [comment]
Just a ping from 2024, is there any progress / changes in creating a MAUI app that runs on a Pi? :)

**Comment by Lug16** [comment]
Another ping from 2026, any progress?

**Reply by holecekp** [reply]
I'd also like to see MAUI running on Linux. These articles gave me a bit of hope even though it will take time:

- https://avaloniaui.net/blog/net-maui-is-coming-to-linux-and-the-browser-powered-by-avalonia
- https://avaloniaui.net/blog/avalonia-maui-progress-update
